client/state/index.js

```diff
@@ -111,7 +114,40 @@ export const reducer = combineReducers( {
 	ui,
 	users,
 	wordads,
-} );
+};
+
+let reducers = { ...initialReducers };
+let onChange = null;
+
+// TODO: See if this export can be eliminated.
+export let reducer = combineReducers( reducers );
+
+export function addReducer( name, reducerFunc ) {
+	if ( reducerExists( name ) ) {
+		throw new Error( 'addReducer(): name "' + name + '" already in use' );
+	}
+
+	reducers = { ...reducers, [ name ]: reducerFunc };
+
+	if ( onChange ) {
+		onChange();
+	}
+}
+
+export function removeReducer( name ) {
+	const { [ name ]: removedReducer, ...remainingReducers } = reducers;
+
+	if ( ! removedReducer ) {
+		log( 'removeReducer(): name not found' );
+		return;
+	}
+
+	reducers = remainingReducers;
+}
+
+export function reducerExists( name ) {
+	return ( reducers[ name ] !== undefined );
+}
```

Comment:
Given that they modify global state, I guess we'd have to export the reducers variable? But that's exactly the global we don't want to mutate directly, right?

I was thinking of going with a class to encapsulate this a bit, but if you have another suggestion, I'm open to it.